### ISSUE TITLE
Potential issue with Rust 2021 edition?

### ISSUE THREAD

**Issue by shapedthought** [FGRibreau/spinners#21]
Hi there, 

I just tried your library on 2021 and it is thowing an error when running the shown example.

thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: NotSupported', C:\Users\exfho\.cargo\registry\src\github.com-1ecc6299db9ec823\spinner-0.5.0\src\lib.rs:164:37

I thought I'd let you know.

**Comment by FGRibreau** [comment]
Hello!

I tried [both example on 2021 edition](https://github.com/FGRibreau/spinners/tree/master/examples) and it worked, could you be more precise about the code you use?

**Comment by sebaslopez01** [comment]
Same with me, I just try it and got the same error

**Comment by FGRibreau** [comment]
What rustc & cargo version do you us?

**Comment by sebaslopez01** [comment]
I'm using 1.58.1, last one that is available

**Comment by sebaslopez01** [comment]
Also I'm in windows 11 system

**Comment by crat0z** [comment]
Are you using VS code to debug? I don't know if this is related, since your panic seems to be a few lines down, but I just had this happen:
```
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/spinner-0.5.0/src/lib.rs:155:44
stack backtrace:
   0:     0x555555fe287c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h583da7c5b1ffcf6e
   1:     0x555556004a6c - core::fmt::write::h776e8fa65c09cfeb
   2:     0x555555fdc005 - std::io::Write::write_fmt::hd8f86d2c4a5cc260
   3:     0x555555fe4410 - std::panicking::default_hook::{{closure}}::hea969ad198ecd943
   4:     0x555555fe3fc5 - std::panicking::default_hook::hd232da7fae64e26b
   5:     0x555555fe4ac4 - std::panicking::rust_panic_with_hook::hffe17de889022375
   6:     0x555555fe4572 - std::panicking::begin_panic_handler::{{closure}}::h9df7a4eeedd84d36
   7:     0x555555fe2d24 - std::sys_common::backtrace::__rust_end_short_backtrace::h50480516fb3ba642
   8:     0x555555fe4509 - rust_begin_unwind
   9:     0x5555556d17a1 - core::panicking::panic_fmt::h92d66d0ceae7016c
  10:     0x5555556d16ed - core::panicking::panic::h2b910859aa32c239
  11:     0x55555587bd91 - core::option::Option<T>::unwrap::h30c8fa47d54d3390
                               at /builddir/build/BUILD/rustc-1.58.1-src/library/core/src/option.rs:746:21
  12:     0x555555878161 - spinner::Spinner::start::{{closure}}::h3c423220c7aadb58
                               at /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/spinner-0.5.0/src/lib.rs:155:29
  13:     0x555555889a6c - std::sys_common::backtrace::__rust_begin_short_backtrace::he109914fc97be8c9
                               at /builddir/build/BUILD/rustc-1.58.1-src/library/std/src/sys_common/backtrace.rs:123:18
  14:     0x55555587d3e1 - std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}::h3983b7e729cd163e
                               at /builddir/build/BUILD/rustc-1.58.1-src/library/std/src/thread/mod.rs:484:17
  15:     0x55555587a751 - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h7799a5541db3b904
                               at /builddir/build/BUILD/rustc-1.58.1-src/library/core/src/panic/unwind_safe.rs:271:9
  16:     0x5555558860bc - std::panicking::try::do_call::hef27b6fb4df3b777
                               at /builddir/build/BUILD/rustc-1.58.1-src/library/std/src/panicking.rs:406:40
  17:     0x55555588646b - __rust_try
  18:     0x555555885ff7 - std::panicking::try::h9118476235622de2
                               at /builddir/build/BUILD/rustc-1.58.1-src/library/std/src/panicking.rs:370:19
  19:     0x555555881961 - std::panic::catch_unwind::h4c8e5b1c42ca802d
                               at /builddir/build/BUILD/rustc-1.58.1-src/library/std/src/panic.rs:133:14
  20:     0x55555587d201 - std::thread::Builder::spawn_unchecked::{{closure}}::h54b1ca9fb550edf4
                               at /builddir/build/BUILD/rustc-1.58.1-src/library/std/src/thread/mod.rs:483:30
  21:     0x5555558761ef - core::ops::function::FnOnce::call_once{{vtable.shim}}::h5e57700ef9eb37c8
                               at /builddir/build/BUILD/rustc-1.58.1-src/library/core/src/ops/function.rs:227:5
  22:     0x555555fe7b33 - std::sys::unix::thread::Thread::new::thread_start::h9141ae0331cdd327
  23:     0x7ffff79a8b1a - start_thread
  24:     0x7ffff7a2d650 - __clone3
  25:                0x0 - <unknown>
```

Placing breakpoint on lib.rs:155 and stepping through comes to mod.rs:63 in TermInfo:
```
impl TermInfo {
    /// Create a `TermInfo` based on current environment.
    pub fn from_env() -> Result<TermInfo> {
        let term_var = env::var("TERM").ok();
        ...
        if let Some(term_name) = term_name {
            TermInfo::from_name(term_name)
        } else {
            /// errors here
            Err(crate::Error::TermUnset)
        }
    }
```

To test your terminal:
```
use std::env;

match env::var("TERM").ok() {
    Some(_) => println!("found term"),
    None => println!("didn't find term"),
}
```

**Comment by cjgratacos** [comment]
Having the same issue.

**Comment by jamesinaxx** [comment]
I was running into this issue too, so I created [a fork](https://github.com/jamesinaxx/spinners-rs) that fixed this issue without reliance on the other spinner crate

**Comment by FGRibreau** [comment]
@jamesinaxx nice ! Interested in merging it back to upstream? :)

**Comment by jamesinaxx** [comment]
@FGRibreau I'd love to. You might wanna look over it first I've made a lot of changes

**Comment by FGRibreau** [comment]
Great!
Fixed in v3.0.1 :+1:
